### ISSUE TITLE
Empty object type returned by flattening a Variant Schema

### ISSUE THREAD

**Issue by ShlokDesai33** [fabian-hiller/valibot#700]
When I try to safe parse with a Variant schema, the nested issues have type = `{} | undefined`. Here's an example:

```ts
const VariantSchema = v.variant('type', [
	v.object({
		type: v.literal('email'),
		email: v.pipe(v.string(), v.email()),
	}),
	v.object({
		type: v.literal('url'),
		url: v.pipe(v.string(), v.url()),
	}),
	v.object({
		type: v.literal('date'),
		date: v.pipe(v.string(), v.isoDate()),
	}),
]);

const { success, issues } = v.safeParse(VariantSchema, { hi: 'hello ' });

if (!success) {
	const { nested, root } = v.flatten<typeof VariantSchema>(issues);
}
```

Is there a fix / workaround to this issue?

**Comment by fabian-hiller** [comment]
Yes, this is a known issue that will be fixed in the future. Currently there is a circular dependency that I need to investigate. See the code [here](https://github.com/fabian-hiller/valibot/blob/v0.36.0/library/src/types/issue.ts#L329-L339).

**Comment by ShlokDesai33** [comment]
You don't need to apologize, this library is the best validation tool out there! Thank you for the workaround!

**Comment by fabian-hiller** [comment]
> When I switch the variant schema to a union of objects, the nested errors is undefined

In my case they are optional undefined because we have no guarantee that every path has an issue. Please take a look at [this playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgDVkphlWYAZSYALAKYhkcALyYAdAFdWwNgAoA2vQCQGORAoArMUxiqA3tq0wAnmDEAueeRhioZVYQnJgpQgEoAGksvH0ddMGA7VV0OXlYAc1VA+RDSJMDtAF8MnT1DY1MLLStbBydgFzc0wgUoXxytWtIwuQiomJg4xOTdJvSgrWyB3X0jE3NLGzsW51d3QgATZBd-Aa0llxa2sWi5WOAEpID5YFQIABFlnb8cofoAXT8GZjYOOCgxVAVSeFkY5GwYgACjxUDtuLx+EJRF5jmY4CJgI5COJSKQIEQ4NlnsBsHBVB8vj85Kcvp8-HAiix2PBsKRlgBRKB4KDoP5yOnLFysAA8UzEEDxEL4AmE4kkAD4CZ9vjASagyagnvRMkA) and let me know if it looks good.

> [...] the root error = 'Invalid type: Expected Object but received Object'

I do not have a better solution for this at the moment. This is a rare case, as you would normally use `variant` for objects.

**Comment by fabian-hiller** [comment]
> When I switch the variant schema to a union of objects, the nested errors is undefined

In my case they are optional undefined because we have no guarantee that every path has an issue. Please take a look at [this playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgDVkphlWYAZSYALAKYhkcALyYAdAFdWwNgAoA2vQCQGORAoArMUxiqA3tq0wAnmDEAueeRhioZVYQnJgpQgEoAGksvH0ddMGA7VV0OXlYAc1VA+RDSJMDtAF8MnT1DY1MLLStbBydgFzc0wgUoXxytWtIwuQiomJg4xOTdJvSgrWyB3X0jE3NLGzsW51d3QgATZBd-Aa0llxa2sWi5WOAEpID5YFQIABFlnb8cofoAXT8GZjYOOCgxVAVSeFkY5GwYgACjxUDtuLx+EJRF5jmY4CJgI5COJSKQIEQ4NlnsBsHBVB8vj85Kcvp8-HAiix2PBsKRlgBRKB4KDoP5yOnLFysAA8UzEEDxEL4AmE4kkAD4CZ9vjASagyagnvRMkA) and let me know if it looks good.

> [...] the root error = 'Invalid type: Expected Object but received Object'

I do not have a better solution for this at the moment. This is a rare case, as you would normally use `variant` for objects.

**Comment by ShlokDesai33** [comment]
The example provided in the playground is exactly the behavior I was looking for! Closing this issue for now since I have implemented the work around you provided. Thank you for the help!

**Comment by fabian-hiller** [comment]
This is fixed in v0.39.0
